prepare.php

```diff
@@ -104,6 +103,58 @@
 $contents = str_replace( array_keys( $search_replace ), array_values( $search_replace ), $contents );
 file_put_contents( $WPT_PREPARE_DIR . '/wp-tests-config.php', $contents );
 
+// Now, install PHPUnit based on the test environment's PHP Version
+$php_version_cmd = $WPT_PHP_EXECUTABLE . " -r \"print PHP_MAJOR_VERSION . '.' . PHP_MINOR_VERSION . '.' . PHP_RELEASE_VERSION;\"";
+if ( ! empty( $WPT_SSH_CONNECT ) ) {
+	$php_version_cmd = 'ssh ' . $WPT_SSH_OPTIONS . ' ' . escapeshellarg( $WPT_SSH_CONNECT ) . ' ' . escapeshellarg( $php_version_cmd );
+}
+
+$retval = 0;
+$env_php_version = exec( $php_version_cmd, $output, $retval );
+if ( $retval !== 0 ) {
+	error_message( 'Could not retrieve the environment PHP Version.' );
+}
+log_message( "Environment PHP Version: $env_php_version" );
+
+// If PHP Version is 8.X.X, set PHP Version to 7.4 for compatibility with core PHPUnit tests.
+if ( substr( $env_php_version, 0 , 2 ) === '8.' ) {
+	log_message( 'Version 8.x.x Found. Downloading PHPUnit for PHP 7.4 instead for compatibility.' );
+	$env_php_version = '7.4';
+}
+
+// If PHP version is 5.6-7.0, download PHPUnit 5.7 phar directly.
```

Comment:
This doesn't check for a lower bound. Should it give an error if the PHP version is lower than 5.6?

Reply:
Done in new commit.